### DISCUSSION TITLE
Academic question: does the runtime really need properties and events?

### DISCUSSION THREAD

**Discussion by TahirAhmadov** [dotnet/runtime#83741] [General]
I understand languages like C# and VB.NET can and probably want to provide these features. However, was it necessary to add the support for these member types to the runtime itself? Couldn't have the language (compiling for a runtime without properties/events) just created the accessor methods directly, and decorated them with some attributes, to provide the necessary language features? Similarly, these attributes could have also been used to enumerate "properties", similarly to `Type.GetProperties` etc. methods. Am I way off base here?

Obviously, this is **not** a request to get rid of properties/events HAHAHA

**Comment by huoyaoyuan** [comment]
> However, was it necessary to add the support for these member types to the runtime itself?

In my sight of view, they are supported by metadata representation, not the execution engine.

**Reply by TahirAhmadov** [reply]
Right, I guess my question was, do we even need those as explicit metadata? I'm curious because the existence of such metadata imposes a requirement that all .NET languages have certain features, and without it, there would be a little more flexibility for language designers.

**Reply by huoyaoyuan** [reply]
Flexibility is in contrast with consistency. Since such pattern is used very frequently in reflection and other common helpers, make it consistent would help a lot.
